src/Razor/test/Microsoft.VisualStudio.LanguageServerClient.Razor.Test/HtmlCSharp/HoverHandlerTest.cs

```diff
@@ -0,0 +1,367 @@
+            var documentMappingProvider = Mock.Of<LSPDocumentMappingProvider>();
+            var hoverhandler = new HoverHandler(JoinableTaskContext, requestInvoker, documentManager, projectionProvider, documentMappingProvider);
+            var hoverRequest = new TextDocumentPositionParams()
+            {
+                TextDocument = new TextDocumentIdentifier() { Uri = Uri },
+                Position = new Position(0, 1)
+            };
+
+            // Act
+            var result = await hoverhandler.HandleRequestAsync(hoverRequest, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task HandleRequestAsync_ProjectionNotFound_ReturnsNull()
+        {
+            // Arrange
+            var documentManager = new TestDocumentManager();
+            documentManager.AddDocument(Uri, Mock.Of<LSPDocumentSnapshot>());
+            var requestInvoker = Mock.Of<LSPRequestInvoker>();
+            var projectionProvider = Mock.Of<LSPProjectionProvider>();
+            var documentMappingProvider = Mock.Of<LSPDocumentMappingProvider>();
+            var hoverhandler = new HoverHandler(JoinableTaskContext, requestInvoker, documentManager, projectionProvider, documentMappingProvider);
+            var hoverRequest = new TextDocumentPositionParams()
+            {
+                TextDocument = new TextDocumentIdentifier() { Uri = Uri },
+                Position = new Position(0, 1)
+            };
+
+            // Act
+            var result = await hoverhandler.HandleRequestAsync(hoverRequest, new ClientCapabilities(), CancellationToken.None).ConfigureAwait(false);
+
+            // Assert
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public async Task HandleRequestAsync_CSharpProjection_InvokesCSharpLanguageServer()
+        {
+            // Arrange
+            var called = false;
+
+            var expectedContents = new SumType<SumType<string, MarkedString>, SumType<string, MarkedString>[], MarkupContent>(
+                new MarkedString()
+                {
+                    Language = "markdown",
+                    Value = "Hover Details"
+                }
+            );
+
+            var lspResponse = new Hover()
+            {
+                Range = new LanguageServer.Protocol.Range()
+                {
+                    Start = new Position(10, 0),
+                    End = new Position(10, 1)
+                },
+                Contents = expectedContents
+            };
+
+            var expectedItem = new Hover()
+            {
+                Range = new LanguageServer.Protocol.Range()
+                {
+                    Start = new Position(0, 0),
+                    End = new Position(0, 1)
+                },
+                Contents = expectedContents
+            };
+
+            var hoverRequest = new CompletionParams()
```

Comment:
CompletionParams?

Reply:
Haha, good catch! Should be `TextDocumentPositionParams`.

When it gets serialized to JSON it's the same structure, so the tests still passed 😆. Updated.